### PR TITLE
[manuf] generalize cert keygen and decouple DICE and TPM cert gen libs

### PR DESCRIPTION
**Pull request by timothytrippel**
This decouples the DICE and TPM key and certificate generation code into separate libs, and make the keymgr derived ECC keygen more generic so SKU owners can provision additional keys and certificates to meet their SKU's requirements. This partially addresses #23426.

### CODE REVIEW

**Comment by timothytrippel** [comment]
CC: @tommychiu-github

File: sw/device/silicon_creator/lib/cert/cert.c [hunk 1]
```diff
@@ -5,14 +5,55 @@
...
+  // output from OTBN in little endian format, so we convert the key to
+  // big endian format.
+  curr_pubkey_le_to_be_convert(pubkey);
+
+  // Generate the key ID.
+  //
+  // Note: the certificate generation functions expect the digest to be in big
+  // endian form, but the HMAC driver returns the digest in little endian, so we
+  // re-format it.
+  hmac_sha256(pubkey, kEcdsaP256PublicKeyCoordBytes * 2, pubkey_id);
```
Comment (cfrantz) [inline sw/device/silicon_creator/lib/cert/cert.c:51]: Considering that `pubkey` is the struct containing the X and Y values, and nothing else, you could use sizeof here:
```
hmac_sha256(pubkey, sizeof(*pubkey), pubkey_id);
util_le_be_buf_format(pubkey_id, sizeof(*pubkey_id));
```

**Review by cfrantz — APPROVED** [review]

File: sw/device/silicon_creator/lib/cert/BUILD [hunk 2]
```diff
@@ -93,3 +99,47 @@ cc_test(
         "@googletest//:gtest_main",
     ],
 )
+
+cc_library(
+    name = "dice",
+    srcs = ["dice.c"],
+    hdrs = ["dice.h"],
+    target_compatible_with = [OPENTITAN_CPU],
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply sw/device/silicon_creator/lib/cert/BUILD]: fixed.

File: sw/device/silicon_creator/lib/base/util.h [hunk 3]
```diff
@@ -28,6 +29,14 @@ uint32_t util_round_up_to(uint32_t input, uint32_t align_bits);
  */
 uint32_t util_size_to_words(uint32_t bytes);
 
+/**
+ * Converts a buffer of bytes from little to big endian in place.
+ *
+ * @param[inout] buf Buffer of in little endian order.
+ * @param num_bytes Number of bytes in the buffer above.
+ */
+void util_le_be_buf_format(unsigned char *buf, size_t num_bytes);
```
[the remark this answers is not in the document]
Comment (timothytrippel) [reply sw/device/silicon_creator/lib/base/util.h]: fixed.

File: sw/device/silicon_creator/lib/cert/cert.c (hunk 1 above)
Comment (timothytrippel) [reply sw/device/silicon_creator/lib/cert/cert.c]: fixed.

**Pull request merged**